### PR TITLE
ramips: Add support for VoCore2

### PR DESCRIPTION
**Pull request by ndoo**
Signed-off-by: Andrew Yong <me@ndoo.sg>

### CODE REVIEW

File: target/linux/ramips/dts/VOCORE2.dts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+&spi0 {
+	status = "okay";
+
+	m25p80@0 {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		linux,modalias = "m25p80";
```
Comment (mkresin) [inline target/linux/ramips/dts/VOCORE2.dts]: drop this line. linux,modalias was used for < kernel 4.4

File: target/linux/ramips/dts/VOCORE2.dts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		partition@50000 {
+			label = "firmware";
+			reg = <0x50000 0xfb0000>;
+		};
+	};
+};
+
+&esw {
+	mediatek,portmap = <0x5>;
+	mediatek,portdisable = <0x3a>;
```
Comment (mkresin) [inline target/linux/ramips/dts/VOCORE2.dts]: Doesn't look correct to me. in the end you have port 0 and port 2 enabled but as far as I can see, this board has only one ethernet interface. The binary mask of you 0x3a portdisable is 111010, where port0 is the most right bit. If the bit is set for a port, the ports is disabled.

File: target/linux/ramips/dts/VOCORE2.dts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+		partition@50000 {
+			label = "firmware";
+			reg = <0x50000 0xfb0000>;
+		};
+	};
+};
+
+&esw {
+	mediatek,portmap = <0x5>;
```
Comment (mkresin) [inline target/linux/ramips/dts/VOCORE2.dts]: Doesn't look correct to me. According to your binary mask you are setting, port 0 is lan, port 1 is wan and port 2 is lan. But as far as I can see, this board has only one ethernet interface.

File: target/linux/ramips/dts/VOCORE2.dts (hunk 1 above)
Comment (ndoo) [inline target/linux/ramips/dts/VOCORE2.dts]: Ack

File: target/linux/ramips/dts/VOCORE2.dts (hunk 3 above)
Comment (ndoo) [inline target/linux/ramips/dts/VOCORE2.dts]: I would like to understand how to bitmask of "mediatek,portmap" bindings work but could not find documentation on this.

Port 0 is LAN, Port 2 is LAN, other ports are not exposed on PCB.

File: target/linux/ramips/dts/VOCORE2.dts (hunk 2 above)
Comment (ndoo) [inline target/linux/ramips/dts/VOCORE2.dts]: Port 0:
Physically exposed on the dock

Port 2:
Physically exposed as pins on the PCB

Port 1,3,4,5:
Not exposed.

Comment (ndoo) [inline target/linux/ramips/dts/VOCORE2.dts]: http://vocore.io/v2.html

**Comment by ndoo** [comment]
Removed `linux,modalias` line from DTS, for other changes requested see comments. esw port 2 was included in DTS as it is physically accessible from the PCB.

File: target/linux/ramips/dts/VOCORE2.dts (hunk 3 above)
Comment (mkresin) [inline target/linux/ramips/dts/VOCORE2.dts]: > I would like to understand how to bitmask of "mediatek,portmap" bindings work but could not find documentation on this.

```git grep mediatek,portmap``` is your friend. Or grep your build_dir to find the corresponding code.

> Port 0 is LAN, Port 2 is LAN, other ports are not exposed on PCB.

Please use 0x7 as portmap (port 0-2 is lan, no wan).

File: target/linux/ramips/dts/VOCORE2.dts (hunk 2 above)
Comment (mkresin) [inline target/linux/ramips/dts/VOCORE2.dts]: Please add this information to the commit message to make the review easier. The following PR is a really good example for a PR having all the required informations to make the review easy: https://github.com/lede-project/source/pull/611

**Comment by ndoo** [comment]
Improvements made to the commit message

**Comment by mkresin** [comment]
Hey Andrew,

perfect commit message!

as far as I can see, only the "wrong" mediatek,portmap is left. Do you agree that it should be changed to 0x7 (port 0 to 2 is lan, no wan).

**Comment by ndoo** [comment]
You are right, I wanted to test flash it with the change to my VoCore2 first before pushing the commit. I have now tested it and here is the updated commit.

**Comment by mkresin** [comment]
@ndoo you obviously have missed to push the commit. still no change here.

**Comment by ndoo** [comment]
Done now sorry

**Comment by mkresin** [comment]
merged, thanks.

**Pull request closed without merging**